REST/c4Listener.cc

```diff
@@ -38,6 +38,45 @@ string C4Listener::URLNameFromPath(slice pathSlice) {
     return Listener::databaseNameFromPath(FilePath(pathSlice, ""));
 }
 
+namespace {
+    bool isEmptyString(const C4String& c4str) {
+        return !c4str || c4str.size == 0;
+    }
```

Comment:
Instead of this, why not have a `toString` method that gets rid of the repeated tertiary operator?

Reply:
toString may yield empty string. That's what the previous version does. I think Pasin wants NULL for empty string.
By the way, this is not strict json string value whereby empty string is "", and here empty string is  , really empty.

Reply:
Right but if you are making a method here anyway, why not have it return the string `"NULL"` here?

Reply:
I think it's better to separate b/w NULL and empty string. If empty string, the log will print "" and if NULL, the log will print NULL. Also the NULL representation of C4String is { nullptr, 0 }?

Reply:
Let me note: we are not doing json, where a string value is quoted both its binary representation of printed form; we print it for human consumption. So, we will print a C4String in 3 shape:
--------
directory: this/path
directory: ""
directory: NULL
--------
sounds good?

Reply:
**BIG and CLEAR**

Reply:
What is printed:
01:15:08.076671| [Listener]: Listener config: {apis: REST, networkInterface: NULL, port: 0, tlsConfig: {}, httpAuthCallback: NULL, callbackContext: NULL, directory: "", allowCreateDBs: 0, allowDeleteDBs: 0}